### PR TITLE
[Ready] Search UI

### PR DESCRIPTION
**Pull request by charypar**
New search design. 

# To do

- [x] icon toggle (using CSS)
- [x] full width search field
- [x] full width type ahead suggestions
- [x] all results submit button
- [x] test in stream and article

### CODE REVIEW

File: src/js/search.js [hunk 1]
```diff
@@ -19,10 +36,31 @@ module.exports = {
 			}
 		});
 
+		var form = document.querySelector('.next-header-v2__search-form');
+		var toggle = document.querySelector('.next-header-v2__search-toggle');
+		var input = document.querySelector('.next-header-v2__search-form #search-term');
```
Comment (ironsidevsquincy) [inline src/js/search.js]: any chance we can get `js-` prefixed classes in the header, and hook off those instead

File: src/scss/header-v2/_search.scss [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+.next-header-v2__search-toggle {
+	cursor: pointer;
+
+	&:before {
+		content: "";
+		height: 100%;
+		background-color: #000000;
+		width: 1px;
+		display: block;
+		border-right: 1px solid #525151;
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: is this in our palette

File: src/scss/header-v2/_search.scss [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	}
+}
+
+.next-header-v2__search-toggle {
+	cursor: pointer;
+
+	&:before {
+		content: "";
+		height: 100%;
+		background-color: #000000;
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: I'm in two minds about this, for consistency should get `getColor('black')`

File: src/scss/header-v2/_search.scss [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+		width: 1px;
+		display: block;
+		border-right: 1px solid #525151;
+		position: absolute;
+		top: 0;
+	}
+
+	a:after {
+		@include nextIcon(search, #ffffff, 20);
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: `getColor('white')`

File: src/scss/header-v2/_search.scss [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+		display: inline-block;
+	}
+
+	.next-header-v2__search-toggle-text--close {
+		display: none;
+	}
+
+	.next-header-v2__search-checkbox:checked ~ & {
+		a:after {
+			@include nextIcon(cross, #ffffff, 20);
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: `getColor('white')`

File: src/scss/header-v2/_search.scss [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+		border-radius: 0;
+		font-family: inherit;
+		font-size: inherit;
+		color: white;
+
+		&:focus,
+		&:active {
+			outline: none;
+			border: 0;
+			box-shadow: inset 0 0 3px 1px #525151;
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: colour palette?

File: src/scss/header-v2/_search.scss [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+	.typeahead--active {
+		display: block;
+	}
+
+	.typeahead__item {
+		padding: 7px 5px 7px 60px;
+	}
+
+	.typeahead__link {
+		color: #333333;
```
Comment (ironsidevsquincy) [inline src/scss/header-v2/_search.scss]: &tc - few colours which are repeated - should they be in our palette? if not, define the colour in this project and use it

File: src/js/search.js (hunk 1 above)
Comment (charypar) [inline src/js/search.js]: Yep, good point.

**Comment by ironsidevsquincy** [comment]
Cool :+1: 

Can we get a screenshot in the ticket, for posterity
Also, how does it look in IE8, etc?

**Comment by charypar** [comment]
search
search2
search3

**Comment by charypar** [comment]
Works on IE9 now, will just check on article page again.

**Pull request merged**